src/components/BookmarksList.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { Pressable, StyleSheet, Text, View } from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
+
+import Colours from '../themes/colours'
+import { Folder, FolderItem, checkIsFolder } from '../types/bookmarks'
+
+const styles = StyleSheet.create( {
+  BookmarkContainer: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: 5,
+    backgroundColor: Colours.White,
+  },
+} )
+
+type BookmarkListItemProps = {
+  item: Folder | FolderItem,
+  isFolder: boolean,
+  onPress: () => void,
+}
+
+const BookmarkListItem = ( {
+  item,
+  isFolder,
+  onPress,
+}: BookmarkListItemProps ) => (
+  <Pressable style={styles.BookmarkContainer} onPress={onPress}>
+    <Text>{item?.name ?? item}</Text>
```

Comment:
issue with `.name` here

Reply:
Issue comes from effectively two types spawning from same expression, string or object.

I'd also actually just structure bookmarks like this:

```
type NavigationItem = {
  name: string,
  destinations?: NavigationItem[],
}
```

and so you know if `navigationItem.destinations` is undefined, you've reached the end of drilling into folders and it's a list item.

This approach also is for writing a generic structure for navigation.